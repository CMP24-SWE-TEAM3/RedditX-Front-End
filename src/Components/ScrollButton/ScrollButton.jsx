// imports
import React, {useState} from 'react';
import {ButtonStyled} from './ScrollButton.styled';

/**
 *  Component  which has been showed after 200 px from top page to get back to new feeds
 * @returns {React.Component}
 */
const ScrollButton = () => {

    /**
     * state for the scroll button state to make sure is visible when scrolling or not
     */
    const [visible, setVisible] = useState(false)

    /**
     * function to be called when the pixel position changes and get called by event listeners
     */
    const toggleVisible = () => {
        const scrolled = document.documentElement.scrollTop;
        if (scrolled > 200) {
            setVisible(true)
        } else if (scrolled <= 200) {
            setVisible(false)
        }
    };

    const scrollToTop = () => {
        window.scrollTo({
            top: 0,
            behavior: 'smooth'
        });
    };

    window.addEventListener('scroll', toggleVisible);

    return (
        <ButtonStyled id={'back-to-top-button'} onClick={scrollToTop} show={visible}>Back to Top</ButtonStyled>

    );
}

export default ScrollButton;
