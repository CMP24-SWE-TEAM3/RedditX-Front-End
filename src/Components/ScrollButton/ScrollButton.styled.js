import styled from 'styled-components';

export const ButtonStyled = styled.button`
  display: ${({show}) => show ? 'inline' : 'none'};

  font-family: Noto Sans, Arial, sans-serif;
  font-size: 14px;
  font-weight: 700;
  letter-spacing: unset;
  line-height: 17px;
  text-transform: unset;
  min-height: 32px;
  border: none;
  min-width: 32px;
  padding: 4px 16px;
  align-items: center;
  border-radius: 9999px;
  box-sizing: border-box;
  justify-content: center;
  text-align: center;
  width: auto;
  cursor: pointer;
  color: ${({theme}) => theme.background.primary};
  //border-color: ${({theme}) => theme.color.secondary};
  background-color: ${({theme}) => theme.color.secondary};
  //width: 112px;

  :hover, :active {
    background-color: ${({theme}) => theme.background.hover_background_button_blue};
  }
`
