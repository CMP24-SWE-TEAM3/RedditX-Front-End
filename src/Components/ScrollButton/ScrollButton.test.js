import {shallow} from 'enzyme';

// import Components
import ScrollButton from "./ScrollButton";

describe("Scroll", () => {
    it("this is a test for Scroll Button", () => {
        expect(shallow(<ScrollButton/>)).toMatchSnapshot();

    });
});