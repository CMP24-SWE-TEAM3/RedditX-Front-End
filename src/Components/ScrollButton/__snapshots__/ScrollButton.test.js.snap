// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scroll this is a test for Scroll Button 1`] = `ShallowWrapper {}`;
